hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/BulkInsertDataSetWriter.java

```diff
@@ -182,29 +181,44 @@ public TxnView getTxn(){
         return getResultSet();
     }
 
-
     /**
      * This option is for calculating split row keys in HBase format, provided that split row keys are
      * stored in a csv file
      */
     private void doOutputKeysOnly(Activation activation, TableDescriptor td,
                                   ConglomerateDescriptorList conglomerateDescriptorList, List<Long> allCongloms,
-                                  ArrayList<DDLMessage.TentativeIndex> tentativeIndexList) throws StandardException
+                                  ArrayList<DDLMessage.TentativeIndex> tentativeIndexList)
+            throws StandardException
     {
+        long indexConglom = -1;
+        boolean isUnique = false;
         for (ConglomerateDescriptor searchCD : conglomerateDescriptorList) {
             if (searchCD.isIndex() && !searchCD.isPrimaryKey() && indexName != null &&
                     searchCD.getObjectName().compareToIgnoreCase(indexName) == 0) {
+                indexConglom = searchCD.getConglomerateNumber();
+                DataValueDescriptor dvd =
+                        td.getColumnDescriptor(
+                                searchCD.getIndexDescriptor().baseColumnPositions()[0]).getDefaultValue();
                 DDLMessage.DDLChange ddlChange = ProtoUtil.createTentativeIndexChange(txn.getTxnId(),
                         activation.getLanguageConnectionContext(),
                         td.getHeapConglomerateId(), searchCD.getConglomerateNumber(),
-                        td, searchCD.getIndexDescriptor(), td.getDefaultValue(searchCD.getIndexDescriptor().baseColumnPositions()[0]));
+                        td, searchCD.getIndexDescriptor(),
+                        td.getDefaultValue(searchCD.getIndexDescriptor().baseColumnPositions()[0]));
+                isUnique = searchCD.getIndexDescriptor().isUnique();
                 tentativeIndexList.add(ddlChange.getTentativeIndex());
                 allCongloms.add(searchCD.getConglomerateNumber());
                 break;
             }
         }
+        RowAndIndexGenerator rowAndIndexGenerator = getRowAndIndexGenerator(tentativeIndexList);
+        DataSet rowAndIndexes = dataSet.flatMap(rowAndIndexGenerator);
+        DataSet keys = rowAndIndexes.mapPartitions(
+                new RowKeyGenerator(bulkImportDirectory, heapConglom,
+                        indexConglom, isUnique));
+        keys.collect();
```

Comment:
Is this block of code necessary? It seems that `keys` is not used, and there is no obvious desired side-effect. Could you please check?

Reply:
this code is in the old java file as well, see https://github.com/splicemachine/spliceengine/pull/5451/files#diff-b17ac7253e10a871a3e0c8b4c29689c7cfa082c6910cd30c64d6941697392c42L179-L185 (BulkInsertDataSetWriter.java, lines 179-185). Spotbugs was complaining about `List<String> files` not used, so i removed that code, but turned out this made `com.splicemachine.derby.impl.load.HBaseBulkLoadIT#testComputeTableSplitKeys` fail.

If i understand correctly the `RowKeyGenerator` (line 216) is really generating some data, so `keys.collect()` is really doing something, so i basically reverted this to the state it was before, but moved it to a function.

So yes, this is necessary, and `keys` is used as `keys.collect()` is performing work (writes the row keys).

Reply:
I see, this is interesting, thanks for explaining the details.